pages/_app.js

```diff
@@ -5,16 +5,30 @@ import collections from "@/assets/collections.json";
 import useLocalStorageState from "use-local-storage-state";
 import { uid } from "uid";
 import { useState } from "react";
+import ToastMessage from "@/components/ToastMessage/ToastMessage";
+import MarkAsCorrect from "@/public/icons/MarkAsCorrect.svg";
 
 export default function App({ Component, pageProps }) {
   const [flashcards, setFlashcards] = useLocalStorageState("flashcards", {
     defaultValue: initialFlashcards,
   });
 
+  const [toastMessage, setToastMessage] = useState(null);
   const [currentFlashcard, setCurrentFlashcard] = useState(null);
 
   const [actionMode, setActionMode] = useState("default");
 
+  function showToastMessage(message, variant, icon) {
+    setToastMessage({ message, variant, icon });
+    setTimeout(() => {
```

Comment:
Consider using the useEffect hook to manage the timeout, ensuring it clears properly.